.github/workflows/sphinx_build.yml

```diff
@@ -0,0 +1,34 @@
+name: documentation
+
+on:
+  push:
+    branches:
+    - main
+  pull_request:
+    branches:
+    - main
+
+jobs:
+  docs:
+    name: Build documentation
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+
+        # this action updates pip, setuptools, and wheel before install
+      - name: Install Python dependencies
+        uses: py-actions/py-dependency-install@v4
+        with:
+           path: "docs/requirements.txt"
```

Comment:
```suggestion
        run: pip install .[dev]
```

Reply:
In additiona we would need to add:
```
sphinx
sphinx-rtd-theme
```
to setup.py under `extras_require={'dev':}`